beacon-chain/powchain/deposit.go

```diff
@@ -12,6 +13,12 @@ func (s *Service) processDeposit(ctx context.Context, eth1Data *ethpb.Eth1Data,
 	if err := s.preGenesisState.SetEth1Data(eth1Data); err != nil {
 		return err
 	}
-	s.preGenesisState, err = blocks.ProcessPreGenesisDeposits(ctx, s.preGenesisState, []*ethpb.Deposit{deposit})
-	return err
+	beaconState, err := blocks.ProcessPreGenesisDeposits(ctx, s.preGenesisState, []*ethpb.Deposit{deposit})
```

Comment:
given that we pass in a pointer the state will still be modified here. So this change doesn't help much.